### ISSUE TITLE
`SelectPanel` anchors should have `aria-haspopup="listbox"`

### ISSUE THREAD

**Issue by iansan5653** [primer/react#3629]
### Description

The anchor for [`SelectPanel`](https://primer.style/react/SelectPanel) is rendered with `aria-haspopup="true"` by default. The `true` value for this HTML attribute is equivalent to `menu`, implying that the related popup will have the `menu` role.

`SelectPanel`s have the `listbox` role, not `menu`, so the value of `aria-haspopup` here should be `listbox`.

### Steps to reproduce

Render a `SelectPanel` without a `renderAnchor` prop defined.

### Version

v35.27.1

### Browser

Chrome

**Comment by joshblack** [comment]
Thanks for opening this up @iansan5653! Talking with @ericwbailey on Friday this week about this, will follow up here with the results of that call 👀

**Comment by joshblack** [comment]
One quick clarification, would it be accurate to say that the role whose visibility is being triggered is `dialog` in the `SelectPanel` case? In other words, the. anchor which has `aria-expanded` and `aria-haspopup` is triggering the visibility of a dialog which contains the input and corresponding listbox.

**Comment by iansan5653** [comment]
Hmm I see that. I guess it would be `dialog` then.

**Comment by ichelsea** [comment]
Yeah, selectpanel is _not_ a semantic menu and can't be. It's held within a semantic dialog.

**Comment by joshblack** [comment]
It seems like a change from `aria-haspopup="true"` to `aria-haspopup="dialog"` is mostly innocuous here, is that correct? For screen readers that support it (VoiceOver, JAWS) it will announce as having a dialog. For screen readers that don't (NVDA is the big one), it will fallback to menu button which it currently has due to the implicit role of `aria-haspopup="true"`.

Based on that, moving from `aria-haspopup="true"` to `aria-haspopup="dialog"` for the current implementation of `SelectPanel` seems to work as an enhancement. Does that track with what you're thinking @ichelsea and @ericwbailey? 👀

**Comment by ichelsea** [comment]
@joshblack, yes we want users to be aware that they are in fact opening a dialog and not a menu because they have different expectations of use.
